### PR TITLE
Change refdocs portion of the release process

### PR DESCRIPTION
**Pull request by tmatsuo**
I already put a new `index.html` in the gh-pages branch with links to googleapis.dev.

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2801)

### CODE REVIEW

File: doc/cutting-a-release.md [hunk 1]
```diff
@@ -91,27 +91,14 @@ git push --set-upstream origin "${RELEASE}.x"
...
-Change the value of the meta tag's `URL=` parameter to refer to the
-`index.html` file in the directory matching the new release version number.
-
-```
-git commit -am "Update documentation to ${RELEASE}"
-git push
-```
+## Generate and upload the documentation to googleapis.dev
+
+Manually run a kokoro job
```
Comment (coryan) [inline doc/cutting-a-release.md]: nit: should that be Kokoro (capitalized)?

File: doc/cutting-a-release.md [hunk 2]
```diff
@@ -91,27 +91,14 @@ git push --set-upstream origin "${RELEASE}.x"
...
+`prod:cloud-devrel/client-libraries/cpp/google-cloud-cpp/refdocs` with
```
Comment (coryan) [inline doc/cutting-a-release.md]: Which is found where? I added that build to the Cloud C++ internal dashboard, so maybe just say that?

File: doc/cutting-a-release.md [hunk 3]
```diff
@@ -91,27 +91,14 @@ git push --set-upstream origin "${RELEASE}.x"
...
+specifying the branch name (e.g. `v0.11.x`) as `Committish` field. This job will
```
Comment (coryan) [inline doc/cutting-a-release.md]: s/with specifying/and specify/ maybe
s/as `Committish`/in the `Comittish` field`/ ?

File: doc/cutting-a-release.md [hunk 4]
```diff
@@ -91,27 +91,14 @@ git push --set-upstream origin "${RELEASE}.x"
...
+generate and upload the doxygen documentation to the staging bucket for
+googleapis.dev hosting. The uploaded documentation will be live generally in an
```
Comment (coryan) [inline doc/cutting-a-release.md]: s/will be live generally/will generally be live/ ?

File: doc/cutting-a-release.md (hunk 1 above)
Comment (tmatsuo) [reply doc/cutting-a-release.md]: Done

File: doc/cutting-a-release.md (hunk 2 above)
Comment (tmatsuo) [reply doc/cutting-a-release.md]: Done

File: doc/cutting-a-release.md (hunk 3 above)
Comment (tmatsuo) [reply doc/cutting-a-release.md]: Done

File: doc/cutting-a-release.md (hunk 4 above)
Comment (tmatsuo) [reply doc/cutting-a-release.md]: Done

**Comment by tmatsuo** [comment]
@coryan Thanks! I also fix the branch name detection in the `refdocs.sh`. PTAL

**Pull request merged**